x-pack/legacy/plugins/apm/server/routes/settings.ts

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import { InternalCoreSetup } from 'src/core/server';
+import Joi from 'joi';
+import { setupRequest } from '../lib/helpers/setup_request';
+import { getServiceNames } from '../lib/settings/cm/get_service_names';
+import { saveConfiguration } from '../lib/settings/cm/save_configuration';
+import { updateConfiguration } from '../lib/settings/cm/update_configuration';
+import { CentralConfigurationIntake } from '../lib/settings/cm/configuration';
+import { searchConfigurations } from '../lib/settings/cm/search';
+import { listConfigurations } from '../lib/settings/cm/list_configurations';
+import { getEnvironments } from '../lib/settings/cm/get_environments';
+import { deleteConfiguration } from '../lib/settings/cm/delete_configuration';
+import { createCmIndex } from '../lib/settings/cm/create_cm_index';
+
+const defaultErrorHandler = (err: Error) => {
+  // eslint-disable-next-line
+  console.error(err.stack);
+  throw Boom.boomify(err, { statusCode: 400 });
+};
+
+export function initSettingsApi(core: InternalCoreSetup) {
```

Comment:
Looks like these are all fairly light on request validation. Do we need a little more security here perhaps?